plugins/blueprint.cpp

```diff
@@ -562,27 +586,23 @@ string get_tile_query(df::building* b)
     return " ";
 }
 
-void init_stream(ofstream &out, std::string basename, std::string target)
+void init_stream(ofstream &out, string basename, string target)
 {
     std::ostringstream out_path;
     out_path << basename << "-" << target << ".csv";
     out.open(out_path.str(), ofstream::trunc);
     out << "#" << target << endl;
 }
 
-command_result do_transform(DFCoord start, DFCoord end, string name, uint32_t phases, std::ostringstream &err)
+command_result do_transform(const DFCoord &start, const DFCoord &end,
+                            const blueprint_options &options,
+                            std::ostringstream &err)
 {
     ofstream dig, build, place, query;
 
-    std::string basename = "blueprints/" + name;
-
-#ifdef _WIN32
```

Comment:
Just wanted to confirm that this was intentional. I see you added the normalization in https://github.com/DFHack/dfhack/commit/c19fc1f34903f15e915fa730af6d9ed53f5fc93c#diff-ef1e8e7bfb935197babf6453e5f25e4a208e546ef64fee459c68e03c5df8ebcdR577-R580, presumably for parity with ` Filesystem::mkdir_recursive()`, and likely not a lot of people rely on putting blueprints in subfolders named with backslashes, but there is a slight change in behavior here: given `a/b\c`, the splitting logic (`find_last_of("/")`) would produce `a` as the directory name before passing it to mkdir_recursive. Not a huge deal, IMO.

Reply:
the normalization of name is now handled in the lua code, so it's guaranteed to be in proper format by the time it gets here - see the `parse_positionals()` function

Reply:
Oh, thanks! Missed that.